src/shims/alloc.rs

```diff
@@ -47,6 +47,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 next_pow2 / 2
             }
         }
+        if size == 0 {
+            return Align::from_bytes(1).unwrap();
+        }
```

Comment:
This works, but please move it up above the comment that says we will round down.

Also, you can use `Align::ONE` to avoid an unwrap here.